### PR TITLE
Backports 0.17 pr29

### PR DESCRIPTION
**Pull request by UdjinM6**

### CODE REVIEW

File: src/txmempool.h [hunk 1]
```diff
@@ -640,7 +640,7 @@ class CTxMemPool
     /** Populate setDescendants with all in-mempool descendants of hash.
      *  Assumes that setDescendants includes all in-mempool descendants of anything
      *  already in it.  */
-    void CalculateDescendants(txiter it, setEntries &setDescendants) EXCLUSIVE_LOCKS_REQUIRED(cs);
+    void CalculateDescendants(txiter it, setEntries& setDescendants) const EXCLUSIVE_LOCKS_REQUIRED(cs);;
```
Comment (PastaPastaPasta) [inline src/txmempool.h:643]: remove double semicolon

File: src/qt/addresstablemodel.cpp [hunk 2]
```diff
@@ -404,27 +406,42 @@ bool AddressTableModel::removeRows(int row, int count, const QModelIndex &parent
...
+        return QString::fromStdString(purpose);
+    }
+    return QString();
+}
+
+bool AddressTableModel::getDestinationData(const CTxDestination &dest,
+        std::string* name,
+        std::string* purpose) const {
+    return walletModel->wallet().getAddress(dest, name, /* is_mine= */ nullptr, purpose);
+}
```
Comment (PastaPastaPasta) [inline src/qt/addresstablemodel.cpp:443]: Maybe should add a getAddressData wrapper like you did above? might make future conflicts trivial

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
Couple of thoughts

File: src/qt/addresstablemodel.cpp (hunk 2 above)
Comment (UdjinM6) [reply src/qt/addresstablemodel.cpp:432]: Turned out that `labelForDestination` (which we introduced in #3155) is no longer used anywhere, so I dropped it and re-used the code from 12830. Should be even less merge conflicts in the future now :D

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Review by xdustinface — APPROVED** [review]
utACK

**Pull request merged**